src/holochainClient.js

```diff
@@ -131,14 +133,28 @@ async function initAndGetHolochainClient () {
     if (HOLOCHAIN_LOGGING) {
       console.log('🎉 Successfully connected to Holochain!')
     }
+    isInitiatingHcConnection = false
     return holochainClient
   } catch (error) {
     if (HOLOCHAIN_LOGGING) {
       console.log('😞 Holochain client connection failed -- ', error.toString())
     }
+    isInitiatingHcConnection = false
     throw (error)
   }
 }
+async function initAndGetHolochainClient () {
+  let counter = 0
+  while (isInitiatingHcConnection) {
```

Comment:
can we add a comment calling out that this variable is being changed in a different function running in parallel?